pkg/sql/sqlutil/internal_executor.go

```diff
@@ -191,16 +212,74 @@ type InternalRows interface {
 	Types() colinfo.ResultColumns
 }
 
-// SessionBoundInternalExecutorFactory is a function that produces a "session
+// InternalExecutorProto stores info needed to initialize an
+// internal executor.
+// The only difference between InternalExecutorProto and InternalExecutorFactory
+// is that the proto also includes the synthetic descriptor.
+type InternalExecutorProto struct {
```

Comment:
The word `Proto` here doesn't fit for me. I assume it means protocol, but that's not really a term in go. Proto to me in this codebase always means protobuf. 

I think we should turn this thing either into an interface or into a struct with no exported fields and a number of methods. My preference is an interface so you can implement it elsewhere and not have to do this hacky closure thing we have here now. Just call the interface `InternalExecutorFactory` and get rid of the closure.

Reply:
Ah yeah, I think it makes sense to just keep using the `InternalExecutorFactory`. Actually, the only reason that I made it a struct is to store the `SyntheticDescs`, so that it can be passed to `ieProto.WithTxn()` for the initialization of an internal executor. But I just realize it's not necessary to hang the synthetic descriptors with the factory. Will make the change.